tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -0,0 +1,200 @@
+        ASSERT(is_pointer_interconvertible_base_of_v<A, volatile C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<volatile A, const C>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, D>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A1>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A2>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, A3>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A2, A3>);
+
+        ASSERT(!is_pointer_interconvertible_base_of_v<A1, A2>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A4, A>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<B, C>);
+    }
+
+    // is_corresponding_member tests
+    {
+        struct S1 {
+            int v1;
+            int v2;
+        };
+
+        struct S2 {
+            int v1;
+            int v2;
+        };
+
+        struct S3 {
+            int v1;
+            int v2;
+            int v3;
+        };
+
+        struct S4 {
+            char v1;
+            int v2;
+            int v3;
+        };
+
+        struct S5 {
+            int v1;
+            int v2;
+            void* v3;
+        };
+
+        struct S6 {
+            int v1;
+            int v2;
+            double v3;
+        };
+
+        ASSERT(is_corresponding_member(&S1::v1, &S1::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v1, &S1::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S1::v1));
+        ASSERT(is_corresponding_member(&S1::v1, &S2::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S2::v2));
+        ASSERT(is_corresponding_member(&S1::v1, &S3::v1));
+        ASSERT(is_corresponding_member(&S1::v2, &S3::v2));
+        ASSERT(is_corresponding_member(&S5::v1, &S6::v1));
+        ASSERT(is_corresponding_member(&S5::v2, &S6::v2));
+
+        ASSERT(!is_corresponding_member(&S1::v1, &S2::v2));
+        ASSERT(!is_corresponding_member(&S1::v2, &S2::v1));
+        ASSERT(!is_corresponding_member(&S1::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S1::v2, &S4::v2));
+        ASSERT(!is_corresponding_member(&S3::v1, &S4::v1));
+        ASSERT(!is_corresponding_member(&S3::v2, &S4::v2));
+        ASSERT(!is_corresponding_member(&S5::v1, &S6::v2));
+        ASSERT(!is_corresponding_member(&S5::v2, &S6::v1));
+        ASSERT(!is_corresponding_member(&S5::v3, &S6::v3));
+    }
+
+    // is_pointer_interconvertible_with_class tests
+    {
+        struct A {
+            int a;
+        };
+
+        struct B {
+            int b;
+        };
+
+        struct C : A, B {};
+
+        union U {
+            int v1;
+            char v2;
+        };
+
+        ASSERT(is_pointer_interconvertible_with_class(&C::b));
+        ASSERT(!is_pointer_interconvertible_with_class<C>(&C::b));
+
+        ASSERT(is_pointer_interconvertible_with_class(&U::v1));
+        ASSERT(is_pointer_interconvertible_with_class(&U::v2));
+    }

```

Comment:
WG21-N4878 [meta.member]/2:

> *Returns:* `true` if and only if `S` is a standard-layout type, `M` is an object type, `m` is not null, and each
object `s` of type `S` is pointer-interconvertible (6.8.3) with its subobject `s.*m`.

Additional tests:

* Non-standard-layout type
* Pointer to member function
* Null pointer